### PR TITLE
Add possibility to clean PyDicom objects

### PR DESCRIPTION
**Pull request by RobinFrcd**
# Description

The point of this PR is to add the possibility to work on PyDicom objects directly, in order to speed up the process. It indeed avoids reading the dicom file from disk twice.

Extracting the `clean_pixel_data` function out of the `DicomCleaner` class also helps when we want to process multiple images in multiple processes without having to instance multiple `DicomCleaner` times (which is slow because it needs to load the recipe from disk).

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project

### CODE REVIEW

File: docs/_docs/getting-started/dicom-pixels.md [hunk 1]
```diff
@@ -244,10 +244,11 @@ and then easily see the pixel operations that will be done:
 and then run clean to perform the actions.
 
 ```python
-client.clean()
+client.clean_pixel_data()
```
Comment (vsoch) [inline docs/_docs/getting-started/dicom-pixels.md:247]: I don't believe you renamed the client function?

File: docs/_docs/getting-started/dicom-pixels.md [hunk 2]
```diff
@@ -257,8 +258,16 @@ cleaner.save_dicom(output_folder=os.getcwd())
 Detect means using the deid recipe to parse headers. You must run detect before you try to clean. If you don't:
 
 ```python
-client.clean()
-WARNING Use <deid.dicom.pixels.clean.DicomCleaner object at 0x7fafb70b9cf8>.detect() to find coordinates first.
+client.clean_pixel_data()
```
Comment (vsoch) [inline docs/_docs/getting-started/dicom-pixels.md:261]: Ditto here.

File: docs/_docs/getting-started/dicom-pixels.md [hunk 3]
```diff
@@ -280,7 +289,7 @@ images to a format that you like. Remember that even with flags, if there are no
 associated with the flag, no changes are done to the image.
 
 ```python
-client.clean()
+client.clean_pixel_data()
```
Comment (vsoch) [inline docs/_docs/getting-started/dicom-pixels.md:292]: And here.

File: docs/_docs/getting-started/dicom-pixels.md [hunk 4]
```diff
@@ -289,7 +298,7 @@ If you need to specify a different header for pixel data (default is `PixelData`
 but you might choose also `FloatPixelData` or `DoubleFloatPixelData` you can do:
 
 ```python
-client.clean(pixel_data_attribute="FloatPixelData")
+client.clean_pixel_data(pixel_data_attribute="FloatPixelData")
```
Comment (vsoch) [inline docs/_docs/getting-started/dicom-pixels.md:301]: And here.

File: docs/_docs/getting-started/dicom-pixels.md [hunk 5]
```diff
@@ -335,7 +344,7 @@ that the fix is correct, and behavior might be changed in future versions of
 pydicom to raise an error. To disable the fix, you can do:
 
 ```python
-client.clean(fix_interpretation=False)
+client.clean_pixel_data(fix_interpretation=False)
```
Comment (vsoch) [inline docs/_docs/getting-started/dicom-pixels.md:347]: And here.

Could you please add an example to the docs of not using the client (if that is the purpose of the PR?) That will also help to explain the use case.

**Review by vsoch — COMMENTED** [review]
I'll need to do a more thorough review after work - I'm open to the idea you've suggested here.

File: docs/_docs/getting-started/dicom-pixels.md (hunk 5 above)
Comment (RobinFrcd) [reply docs/_docs/getting-started/dicom-pixels.md]: https://github.com/pydicom/deid/pull/223/commits/b01a4530a84ab883f4d5fa666682f2970c172127

File: deid/dicom/pixels/clean.py [hunk 6]
```diff
@@ -84,167 +90,22 @@ def detect(self, dicom_file):
...
-            # greyscale image: no need to stack into the channel dim since it doesn't exist
-            elif len(self.original.shape) == 2:
-                self.cleaned = mask * self.original
-
-            else:
-                bot.warning(
-                    "Pixel array dimension %s is not recognized."
-                    % (str(self.original.shape))
-                )
+            return None
```
Comment (vsoch) [inline deid/dicom/pixels/clean.py:99]: Usually you can just return and by default it will return None.

File: deid/dicom/pixels/clean.py [hunk 7]
```diff
@@ -397,3 +258,163 @@ def save_dicom(self, output_folder=None, image_type="cleaned"):
...
+
+    take a dicom image and a list of pixel coordinates, and return
+    a cleaned file (if output file is specified) or simply plot
+    the cleaned result (if no file is specified)
+
+    Parameters
+    ==========
+        dicom_file: (str or FileDataset instance) Dicom file to clean
+        results: Result of the .has_burned_pixels() method
+        fix_interpretation: fix the photometric interpretation if found off
```
Comment (vsoch) [inline deid/dicom/pixels/clean.py:280]: Missing the pixel_data_attribute here, and the parameter should be in line with the title.

File: deid/dicom/types.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+from typing import Union
```
Comment (vsoch) [inline deid/dicom/types.py:1]: Could you please add the same license header to this new file?

File: docs/_docs/getting-started/dicom-pixels.md [hunk 9]
```diff
@@ -341,3 +341,26 @@ client.clean(fix_interpretation=False)
...
+
+There are some cases when using the client won't be handy (multiprocessing) or working on PyDicom files directly would be preferred.
+In that case, you can skip client initialization and do:
+
+```python
+import pydicom
+import matplotlib.pyplot as plt
+
+from deid.dicom.pixels import clean_pixel_data, has_burned_pixels
+
```
Comment (vsoch) [inline docs/_docs/getting-started/dicom-pixels.md:356]: Perfecto! Thanks for adding this. It also helps me to see/understand the use case.

**Review by vsoch — COMMENTED** [review]
Nice work! Let's get tests running. If all is good, the last remaining thing to do is bump the version in deid/version.py, and then write a line in the CHANGELOG.md for it.

File: CHANGELOG.md [hunk 10]
```diff
@@ -14,6 +14,7 @@ and **Merged pull requests**. Critical items to know are:
 Referenced versions in headers are tagged on Github, in parentheses are for pypi.
 
 ## [vxx](https://github.com/pydicom/deid/tree/master) (master)
+ - Add possibility to clean PyDicom objects directly (and without the `DicomClear` if needed) [#223](https://github.com/pydicom/deid/pull/223) (0.2.34)
```
Comment (vsoch) [inline CHANGELOG.md:17]:
```suggestion
 - Add function to clean datasets without `DicomCleaner` [#223](https://github.com/pydicom/deid/pull/223) (0.2.34)
```

**Comment by RobinFrcd** [comment]
@vsoch I think we're good :+1:

**Comment by vsoch** [comment]
I do too! I'm in the middle of a work day so likely I'll do one final look, merge and release this evening. Stay tuned!

**Comment by vsoch** [comment]
Snuck it in :) https://pypi.org/project/deid/0.2.34/ Thank you @RobinFrcd !

**Pull request merged**